devops/scripts/deploy_shared_service.sh

```diff
@@ -49,7 +49,7 @@ if [[ "$last_result" != 0 ]]; then
 fi
 
 deployed_shared_service=$(echo "${get_shared_services_result}" \
-  | jq -r ".sharedServices[] | select(.templateName == \"${template_name}\" and .deploymentStatus == \"deployed\")")
```

Comment:
Yeah this condition is something I missed when I was working on this https://github.com/microsoft/AzureTRE/pull/2235/files 
Sorry!

I think the right condition is to check that the status is not "deleted". As if it is "deleted", we can attempt the deployment again, but not in any other case

Reply:
Actually, if the status is deploymentFailed, we can also attempt the deployment again

Reply:
@tanya-borisova and we won't get a 409 in either of those cases?

Reply:
Made that change.